lib/interfaces/i_repository.dart

```diff
@@ -0,0 +1,4 @@
+abstract class MyRepository<T> {
+
+  Future<List<T>> getData();
```

Comment:
I'd mark the return type of `getData` as `Future<T>`, more generic! Then `LocalRepository` could implement this interface too!

Reply:
Done!